### PR TITLE
WIP: Fix MacOS build

### PR DESCRIPTION
**Pull request by alex-zaitsev**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Short description (up to few sentences):

1. Fixed throwFromErrno invocations for Mac OS that has been broken after recent refactoring.
2. Fixed Elf.h to work on Mac OS
3. Made Parquet optional, so we can build with  -DENABLE_PARQUET=OFF

### CODE REVIEW

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
Tabs -> spaces.

File: dbms/src/Common/Elf.h [hunk 1]
```diff
@@ -6,9 +6,42 @@
 #include <optional>
 #include <functional>
 
+#ifdef __APPLE__
```
Comment (alexey-milovidov) [inline dbms/src/Common/Elf.h]: But Apple uses Mach-O instead of ELF.
This file (along with: Dwarf, SymbolCache, PHDRCache) is totally useless on Mac.

Comment (alex-zaitsev) [reply dbms/src/Common/Elf.h]: Elf.h is included from several files. It is easier to fix a header rather than all of them.

Comment (alexey-milovidov) [reply dbms/src/Common/Elf.h]: Ok.

**Comment by risicle** [comment]
To fix the macos build I suspect you'll need to also look at `libs/libcommon/src/sleep.cpp`, which uses `clock_nanosleep` and `TIMER_ABSTIME`, neither of which are supported on macos from what I can tell.

**Comment by alexey-milovidov** [comment]
(incomplete)

**Comment by Tasselmi** [comment]
++17 -MD -MT libs/libcommon/CMakeFiles/common.dir/src/sleep.cpp.o -MF libs/libcommon/CMakeFiles/common.dir/src/sleep.cpp.o.d -o libs/libcommon/CMakeFiles/common.dir/src/sleep.cpp.o -c ../libs/libcommon/src/sleep.cpp
../libs/libcommon/src/sleep.cpp: In function 'void sleepForNanoseconds(uint64_t)':
../libs/libcommon/src/sleep.cpp:31:40: error: 'TIMER_ABSTIME' was not declared in this scope
     while (clock_nanosleep(clock_type, TIMER_ABSTIME, &finish_time, nullptr) == EINTR);
                                        ^~~~~~~~~~~~~
../libs/libcommon/src/sleep.cpp:31:12: error: 'clock_nanosleep' was not declared in this scope
     while (clock_nanosleep(clock_type, TIMER_ABSTIME, &finish_time, nullptr) == EINTR);
            ^~~~~~~~~~~~~~~
../libs/libcommon/src/sleep.cpp:31:12: note: suggested alternative: 'nanosleep'
     while (clock_nanosleep(clock_type, TIMER_ABSTIME, &finish_time, nullptr) == EINTR);
            ^~~~~~~~~~~~~~~
            nanosleep
[540/3536] Building CXX object contrib/fastops-cm.../fastops.dir/__/fastops/fastops/avx/ops_avx.cpp.o
ninja: build stopped: subcommand failed.


what happened  ?

**Comment by alexey-milovidov** [comment]
@Tasselmi This pull request was abandoned and Mac OS build was not fixed.

**Comment by javisantana** [comment]
is there any plans to make it work again?

**Comment by alexey-milovidov** [comment]
Yes: https://github.com/yandex/ClickHouse/issues/6459

**Comment by alexey-milovidov** [comment]
But you don't have to wait this task: you can continue to fix it and send a PR.

**Pull request merged**